### ISSUE TITLE
Errors syncing IssueTasks from multiple projects from the same provider. Affects gitlab and caldev provides

### ISSUE THREAD

**Issue by ishanarora** [johannesjo/super-productivity#1634]
### Current Behavior

When IssueTasks from multiple projects from the same provider are visible in the same view ("Today", "Timeline", "Schedule"), getFreshDataForIssueTasks gets called for each provider with projects from all the tasks with that provider as the issueType.

This means .e.g issue tasks from multiple gitlab projects may be passed to a single invocation of `GitlabCommonInterfacesService.getFreshDataForIssueTasks`. However, the gitlab and caldev implementations of `getFreshDataForIssueTasks` assume that all tasks belong to the same project, and use the following code to retrieve the projectId:
```
const projectId = tasks && tasks[0].projectId ? tasks[0].projectId : 0;
```

This has some unintentional effects, such as:
1. The provider's api calls may repond with less issues than expected by the provider, causing errors on the stack.
2. Updates from the issues of one project may be synced into the tasks of another project.

### Expected Behavior

`getFreshDataForIssueTasks` implementations of gitlab and caldev shouldn't assume a single project for the tasks, instead they should group the tasks by projects and then proceed with the api calls per group.

Providers other than gitlab and caldev do not face this bug as their implementation of `getFreshDataForIssueTasks` is to simply `map` `getFreshDataForIssueTask` to each task individually. The same approach may be used for gitlab and caldev, although less efficient than bulk updates.

### Your Environment

- Version used: v7.6.0 (should affect every version since v6.0.0)
- Operating System and version: Windows 10 (should affect every operating system)
- Desktop Environment: NA (should affect every desktop environment)
- Browser Name and version: NA (should affect every browser and version)

### Steps to Reproduce (for a crash)

1. Create 2 projects with a gitlab host, and add them to Super Productivity.
2. Create an issue with the same (internal) issue id with each of the projects.
3. Once the 2 issues are available in Super Productivity, add them both to "Today".
4. Open "Today" view and wait for the refresh of these issues.
5. Upon refresh the app crashes.

### Related issues
#1414, #1624

**Comment by johannesjo** [comment]
Thank you very much for reporting this! Would you by any chance be willing to provide a PR for this?

**Comment by ishanarora** [comment]
I started working on a PR for the gitlab issue but I found @cbergmann already fixed this for gitlab in commit 48e2b919f. I can send a PR with the same fix for caldav but I won't be able to test it.

**Comment by johannesjo** [comment]
@ishanarora a PR would be welcome for this. As the code should be functionally similar, the risk of it not working should be low I think.
